helix-core/src/main/java/org/apache/helix/monitoring/mbeans/dynamicMBeans/DynamicMBeanProvider.java

```diff
@@ -103,7 +103,7 @@ protected synchronized boolean doRegister(Collection<DynamicMetric<?, ?>> dynami
    * @param description description of the MBean
    * @param dynamicMetrics the DynamicMetrics
    */
-  private void updateAttributtInfos(Collection<DynamicMetric<?, ?>> dynamicMetrics,
+  public void updateAttributeInfo(Collection<DynamicMetric<?, ?>> dynamicMetrics,
```

Comment:
This might need to be a separate PR.. Let me know if it is neccessary.

Reply:
1. protected?
2. Thanks for fixing the typo, let use "updateAttributtesInfo"?
3. You will need to add synchronized if it is called somewhere else outside this class. Actually we need to add this to all the methods that are reading or updating _attributeMap and _mBeanInfo.
And please don't lock globally, a separate lock for _attributeMap and _mBeanInfo would be good enough.